src/test/java/com/varukha/onlinebookstore/service/category/impl/CategoryServiceImplTest.java

```diff
@@ -0,0 +1,171 @@
+package com.varukha.onlinebookstore.service.category.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.varukha.onlinebookstore.dto.category.CategoryDto;
+import com.varukha.onlinebookstore.exception.EntityNotFoundException;
+import com.varukha.onlinebookstore.mapper.CategoryMapper;
+import com.varukha.onlinebookstore.model.Category;
+import com.varukha.onlinebookstore.repository.category.CategoryRepository;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+@ExtendWith(MockitoExtension.class)
+class CategoryServiceImplTest {
+    @InjectMocks
+    private CategoryServiceImpl categoryService;
+    @Mock
+    private CategoryRepository categoryRepository;
+    @Mock
+    private CategoryMapper categoryMapper;
+
+    @Test
+    @DisplayName("""
+            Test the 'getAll' method with valid request parameters
+            and pagination to retrieve all book categories.
+            """)
+    void getAll_ValidCategoryDtoRequestData_ReturnCategoryDto() {
+        Category category = new Category();
```

Comment:
Please create constant everywhere when you need test data, not only in bookRepositoryTest